### PR TITLE
Added topography box test cases from Akshay's branches

### PR DESCRIPTION
**Pull request by bischtob**
This PR aims to add two mountain test cases to the main branch.

1. Agnesi Mountain (nonhydrostatic + total energy)
2. Schar Mountain (nonhydrostatic + total energy)

This allows other users to access generic box mountain test cases for experimentation with new formulations of the equations.

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by bischtob** [comment]
bors try

File: examples/hybrid/plane/topo_agnesi_nh.jl [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+    @. dρe += hκ₂∇²h_tot
+    @. dρe += vκ₂∇²h_tot
+
+    # Sponge tendency
+    β = @. rayleigh_sponge_z(coords.z)
+    βx = @. rayleigh_sponge_x(coords.x)
+    @. duₕ -= β * (uₕ - u_init)
+    @. dw -= Ic2f(β) * fw
+    #@. duₕ -= βx * (uₕ - u_init)
+    #@. dw -= Ic2f(βx) * fw
```
Comment (valeriabarra) [inline examples/hybrid/plane/topo_agnesi_nh.jl:330]: I would remove stale comments and the related unused `rayleigh_sponge_x` function

**Comment by valeriabarra** [comment]
I can't comments on the artifacts produced, but this seems to be weird Schar Mountain's contravariant3.mp4 seems very weird

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -568,6 +568,28 @@ steps:
...
+      - "julia --color=yes --project=examples examples/hybrid/plane/topo_agnesi_nh.jl"
+    artifact_paths:
+      - "examples/hybrid/plane/output/agnesi_etot_nh/*"
+    agents:
+      config: cpu
+      queue: central
+      slurm_ntasks: 1
+
+  - label: ":computer: Schar Mountain total energy (topography mesh interface)"
+    key: "cpu_schaer_mtn_2d_hybrid_invariant_total_energy_topography"
```
Comment (jiahe23) [inline .buildkite/pipeline.yml:583]:
```suggestion
    key: "cpu_schar_mtn_2d_hybrid_invariant_total_energy_topography"
```

**Comment by bischtob** [comment]
@akshaysridhar, can you please take a look at this?

File: examples/Manifest.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # This file is machine-generated - editing it directly is not advised
 
-julia_version = "1.8.3"
+julia_version = "1.8.2"
```
Comment (jiahe23) [inline examples/Manifest.toml:3]: Let's stay with 1.8.3?

**Review by jiahe23 — APPROVED** [review]
Looks good to me, unless @akshaysridhar wants to take a look before it is merged.

File: examples/hybrid/plane/topo_agnesi_nh.jl [hunk 4]
```diff
@@ -0,0 +1,421 @@
+push!(LOAD_PATH, joinpath(@__DIR__, "..", ".."))
+
```
Comment (simonbyrne) [inline examples/hybrid/plane/topo_agnesi_nh.jl:2]:
```suggestion
```
This can be removed

File: examples/hybrid/plane/topo_schar_nh.jl [hunk 5]
```diff
@@ -0,0 +1,757 @@
+push!(LOAD_PATH, joinpath(@__DIR__, "..", ".."))
+
```
Comment (simonbyrne) [inline examples/hybrid/plane/topo_schar_nh.jl:2]:
```suggestion
```
Same here

**Review by simonbyrne — CHANGES_REQUESTED** [review]
Since both files are pretty similar, can they be combined into one that can be run with different options?

**Comment by bischtob** [comment]
> Since both files are pretty similar, can they be combined into one that can be run with different options?
Let's keep two files for now. We may just delete one later.

**Comment by bischtob** [comment]
bors try

**Pull request merged**